### PR TITLE
feat(skills): add queue and usage-guard usage-window scheduling skills

### PR DESCRIPTION
**Pull request by nikhilsitaram**
## Summary
- Add two macOS-only skills for working around Claude's 5-hour usage window:
  - **queue** — schedule commands to fire later in the same session via a one-shot `CronCreate`, defaulting to ~90s after the current 5h window resets (fresh quota), or at an explicit time/duration. Depends on a thin `statusLine` wrapper that taps `rate_limits.five_hour.{resets_at,used_percentage}` into a state file.
  - **usage-guard** — work a task autonomously, checking 5h consumption at each checkpoint via the same state file, and stop (or `--queue` the remainder into the next block) at a threshold (default 99%).
- Wire both into the `claude-caliper` (full) and `claude-caliper-tooling` bundles; bump bundle versions; document in README.md and CLAUDE.md.
- Both consumers share one state-file contract (`~/.claude/queue/state.json`); the wrapper is the sole producer.

## Implementation review fixes (commit 460b0c0)
A standalone implementation review surfaced four cross-component issues, all fixed:
- Added a producer→consumer integration test (`tests/queue/caliper-test_statusline_seam.sh`) that feeds a real statusline JSON blob through the wrapper, then runs both consumers against the file it actually wrote — closing a seam previously covered only by hand-built fixtures (the wrapper had zero tests).
- Hardened the wrapper to guard `resets_at` to a bare integer epoch before interpolating it into JSON (a non-numeric value can no longer write a corrupt state file).
- Unified the staleness threshold: `check-usage.sh` now emits `STALE=yes|no` on the same >90s cutoff as `compute-fire.sh`; the SKILL.md references that verdict instead of a contradictory "120s".
- Corrected README bundle skill counts (15 / 10 / 5).

## Test plan
- macOS suite (BSD `date`): seam 21, compute-fire 19, check-usage 14 — all pass.
- These skills are intentionally **not** wired into the Linux CI job (they `date -r 0` guard and SKIP on GNU date); the new seam test follows the same pattern.
- CI-relevant tests (hooks, validate-plan) run clean locally.

Co-Authored-By: Claude <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Added two new scheduling-related skills for managing work around Claude usage windows.
  * Introduced support for deferring tasks until a later time and for pausing work before usage limits are reached.
* **Documentation**
  * Expanded setup and usage guidance for the new scheduling workflows.
  * Added notes about platform-specific behavior and required configuration.
* **Tests**
  * Added coverage for scheduling, usage checks, and state-file integration across the new workflows.

### CODE REVIEW

**Review by nikhilsitaram — COMMENTED** [review]
Reviewed the queue / usage-guard bash scripts and their tests purely on their own merits, with focus on shell quoting, exit-code handling, and the producer/consumer state-file contract (statusline-wrapper.sh -> state.json -> compute-fire.sh + check-usage.sh).

I ran all three test suites locally on BSD date: compute-fire 19/19, check-usage 14/14, statusline-seam 21/21 — all pass. I also exercised edge cases the tests don't cover (used_percentage=0, epoch==now, sub-minute future bump, past resets_at) and they all behave correctly.

The code is carefully written: numeric guards on every interpolated field, `10#` octal-safe minute/hour parsing, float-safe awk comparisons, the :00/:30 one-shot-jitter dodge, the DST spring-forward round-trip warning, and a guard so the wrapper never writes corrupt JSON when resets_at is non-numeric (verified end-to-end by the seam test). jq's `//` correctly passes a literal 0 used_percentage through rather than falling to the alternative.

**Issues found:** 0
**Highest severity:** none
**Recommendation:** merge as-is

Minor non-blocking observations (no inline comments warranted): check-usage prints a negative RESETS_IN_MIN when resets_at is in the past — cosmetic only, since used_percentage drives the verdict; and the `!= "null"` resets_at check in check-usage is dead-but-harmless defensiveness (jq `// empty` never emits literal "null").

File: skills/queue/scripts/statusline-wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# JSON below, so a non-numeric value (e.g. an ISO-8601 string) would write a
+# corrupt, unparseable state file — worse than no file, since both consumers
+# would then jq-fail to empty and misreport the cause as "no resets_at".
+resets_at="$(printf '%s' "$input" | jq -r '.rate_limits.five_hour.resets_at // empty' 2>/dev/null)"
+case "$resets_at" in ''|*[!0-9]*) resets_at="" ;; esac
+if [ -n "$resets_at" ]; then
+  used="$(printf '%s' "$input" | jq -r '.rate_limits.five_hour.used_percentage // empty' 2>/dev/null)"
+  now="$(date +%s)"
+  printf '{"resets_at":%s,"used_percentage":%s,"captured_at":%s}\n' \
+    "$resets_at" "${used:-null}" "$now" > "$STATE_FILE"
```
[the remark this answers is not in the document]
Comment (nikhilsitaram) [reply skills/queue/scripts/statusline-wrapper.sh]: Fixed in 0acd377 — `used_percentage` is now sanitized to JSON `null` unless it's a bare number (digits + at most one dot), so a non-numeric value can no longer write corrupt JSON. Same guard class as `resets_at`. Covered end-to-end by the seam test ("non-numeric used_percentage -> null").

File: skills/usage-guard/scripts/check-usage.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+fi
+
+used="$(jq -r '.used_percentage // empty' "$STATE_FILE" 2>/dev/null)"
+resets_at="$(jq -r '.resets_at // empty' "$STATE_FILE" 2>/dev/null)"
+captured="$(jq -r '.captured_at // 0' "$STATE_FILE" 2>/dev/null)"
+if [ -z "$used" ]; then
+  echo "ERROR: no used_percentage in state file (Pro/Max only, after first API response)." >&2
+  exit 2
+fi
+case "$captured" in ''|*[!0-9]*) captured=0 ;; esac
```
[the remark this answers is not in the document]
Comment (nikhilsitaram) [reply skills/usage-guard/scripts/check-usage.sh:47]: Fixed in 0acd377 — `check-usage.sh` now fails **closed** (exit 2) on a missing/non-numeric `used_percentage` rather than letting awk read it as 0 and emit a false `UNDER` (which would let the guard run past the limit). `resets_at` is also guarded to integer-or-empty, and the now-dead `!= "null"` branch was dropped. New tests assert exit 2 on `garbage` and `9.9.9`.

File: skills/queue/scripts/compute-fire.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  if [ ! -f "$STATE_FILE" ]; then
+    echo "ERROR: no state file at $STATE_FILE — the statusline hasn't captured a reset time yet." >&2
+    echo "Fix: keep this terminal focused ~10-15s so the statusline renders once, then retry." >&2
+    exit 1
+  fi
+  resets_at="$(jq -r '.resets_at // empty' "$STATE_FILE" 2>/dev/null)"
+  if [ -z "$resets_at" ] || [ "$resets_at" = "null" ]; then
+    echo "ERROR: state file has no resets_at (rate_limits is Pro/Max-only, and appears after the first API response)." >&2
+    exit 2
+  fi
```
[the remark this answers is not in the document]
Comment (nikhilsitaram) [reply skills/queue/scripts/compute-fire.sh]: Fixed in 0acd377 — `resets_at` is rejected at read with a `case ''|*[!0-9]*` guard (subsuming the old `-z`/`= null` checks) before it can reach `fire=$(( resets_at + 90 ))`. New test asserts a non-integer `resets_at` -> exit 2.

File: skills/queue/scripts/compute-fire.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  # Fire 90s after reset, rounded UP to the next whole minute, dodging :00/:30.
+  # Use the LOCAL minute (date +%M) for the dodge so it stays correct in
+  # half-hour-offset timezones, where UTC minutes != local minutes.
+  fire=$(( resets_at + 90 ))
+  fire=$(( (fire + 59) / 60 * 60 ))
+  m=$(( 10#$(date -r "$fire" +%M) ))
+  if [ "$m" -eq 0 ] || [ "$m" -eq 30 ]; then
+    fire=$(( fire + 60 ))
+  fi
```
[the remark this answers is not in the document]
Comment (nikhilsitaram) [reply skills/queue/scripts/compute-fire.sh:74]: Dismissed — `fire` here is an internally-computed, already-validated positive epoch (`resets_at + 90`, rounded, and `resets_at` is now integer-guarded per the fix above). `date -r "$fire"` on a valid epoch cannot fail or return empty, so the `10#` empty-string case is unreachable; guarding all five `date -r` call sites for an impossible state would add noise without changing behavior.

**Comment by nikhilsitaram** [comment]
## Review feedback addressed (wave 2 — `0acd377`)

**Internal PR reviewer (opus):** 0 issues — merge as-is.

**Gemini Code Assist** — 4 findings, all dispositioned (replies in each thread):

| Finding | Disposition |
|---------|-------------|
| `statusline-wrapper.sh` — `used_percentage` interpolated unvalidated | **Fixed** — sanitized to JSON `null` unless a bare number; same guard class as `resets_at` |
| `check-usage.sh` — non-numeric `used` → false `UNDER` | **Fixed** — fails **closed** (exit 2) instead, so the guard never runs past the limit; `resets_at` also integer-guarded |
| `compute-fire.sh` — `resets_at` unvalidated before arithmetic | **Fixed** — rejected at read via `case` guard before reaching `$(( resets_at + 90 ))` |
| `compute-fire.sh:70` — `date` empty → `10#` error | **Dismissed** — `fire` is an internally-computed, already-validated positive epoch; `date -r "$fire"` cannot fail/return empty here |

**CodeRabbit:** auto-summary only, no actionable findings. **Qodo:** paused for this account (informational).

**Tests:** added non-numeric coverage to all three suites — seam 24, compute-fire 20, check-usage 16, all pass on macOS. CI `test` job green.
